sphinx/search/__init__.py

```diff
@@ -403,10 +403,10 @@ def freeze(self) -> dict[str, Any]:
             for title, titleid in titlelist:
                 alltitles.setdefault(title, []).append((fn2index[docname], titleid))
 
-        index_entries: dict[str, list[tuple[int, str, bool]]] = {}
+        index_entries: dict[str, list[tuple[int, str, Literal[0, 1]]]] = {}
         for docname, entries in self._index_entries.items():
             for entry, entry_id, main_entry in entries:
-                index_entries.setdefault(entry.lower(), []).append((fn2index[docname], entry_id, main_entry == "main"))
+                index_entries.setdefault(entry.lower(), []).append((fn2index[docname], entry_id, 1 if main_entry == "main" else 0))
```

Comment:
```suggestion
                index_entries.setdefault(entry.lower(), []).append((fn2index[docname], entry_id, int(main_entry == "main")))
```

Reply:
`mypy` doesn't seem able to type-narrow that to `Literal[0, 1]`, something I thought would be a nice addition typehint-wise -- so this would be annotated `int` in our code.  I'd be OK with that, though.  What do you reckon?

Reply:
Ahh, I didn't realise. Keep as is.